src/System.Management.Automation/engine/ArgumentToVersionTransformationAttribute.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Management.Automation
+{
+    /// <summary>
+    /// To make it easier to specify a version, we add some conversions that wouldn't happen otherwise:
+    ///   * A simple integer, i.e. 2;
+    ///   * A string without a dot, i.e. "2".
+    /// </summary>
+    internal class ArgumentToVersionTransformationAttribute : ArgumentTransformationAttribute
+    {
+        private static readonly IReadOnlyDictionary<string, Version> _empty = new Dictionary<string, Version>();
+
+        private readonly IReadOnlyDictionary<string, Version> _map;
+
+        /// <summary>
+        /// Initializes a new instance of the ArgumentToVersionTransformationAttribute class with empty version map.
+        /// </summary>
+        public ArgumentToVersionTransformationAttribute() : this(_empty)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the ArgumentToVersionTransformationAttribute class.
+        /// </summary>
+        /// <param name="map">Version mapping.</param>
+        public ArgumentToVersionTransformationAttribute(IReadOnlyDictionary<string, Version> map)
+        {
+            _map = map ?? _empty;
+        }
```

Comment:
I don't think this constructor is useful to an attribute type, because you cannot specify a non-constant value when declaring an attribute to a parameter. This should be removed.